tests/integration/test_katib_experiments.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for Katib Experiments.
+"""
+
+import logging
+
+import lightkube
+import pytest
+from pytest_operator.plugin import OpsTest
+from utils import (
+    assert_deleted,
+    assert_exp_status_running_succeeded,
+    assert_get_experiment,
+    create_experiment,
+    delete_experiment,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "experiment_file",
+    [
+        "tests/assets/crs/bayesian-optimization.yaml",
+        "tests/assets/crs/cmaes.yaml",
+        "tests/assets/crs/darts-cpu.yaml",
+        "tests/assets/crs/enas-cpu.yaml",
+        "tests/assets/crs/file-metrics-collector.yaml",
+        "tests/assets/crs/grid-example.yaml",
+        "tests/assets/crs/hyperband.yaml",
+        "tests/assets/crs/median-stop.yaml",
+        "tests/assets/crs/random.yaml",
+        "tests/assets/crs/simple-pbt.yaml",
+    ],
+)
+async def test_kaitb_experiments(ops_test: OpsTest, experiment_file):
+    """Test Katib Experiments.
+
+    Create experiment and assert that it is Running or Succeeded.
+    At the end, experiment is deleted.
+    NOTE: This test is re-using deployment created in test_deploy_katib_charms() in test_charms.py
+    """
+    namespace = ops_test.model_name
+    lightkube_client = lightkube.Client()
+
+    # Add metrics collector injection enabled label to namespace
+    test_namespace = lightkube_client.get(
+        res=lightkube.resources.core_v1.Namespace, name=namespace
+    )
+    test_namespace.metadata.labels.update(
+        {"katib.kubeflow.org/metrics-collector-injection": "enabled"}
+    )
+    lightkube_client.patch(
+        res=lightkube.resources.core_v1.Namespace,
+        name=test_namespace.metadata.name,
+        obj=test_namespace,
+    )
+
+    exp_name = create_experiment(
+        client=lightkube_client, exp_path=experiment_file, namespace=namespace
+    )
```

Comment:
I won't block on this, but I did feel I had to raise it. 

Katib does have a dedicated client which IMO is way friendlier than lightkube. This would obviously affect the way you create/get resources (in your utils.py as well), but I do think it would make sense for us to leverage their own client, which BTW also offers shortcuts e.g. for checking whether an experiment succeeded. You can see how I use it here: https://github.com/canonical/charmed-kubeflow-uats/blob/main/tests/notebooks/katib-integration.ipynb

Reply:
that's a good observation, I can file an issue to have it as an enhancement so we can move on.

Reply:
Yep, let's do that!

Reply:
filed #115